presto-main/src/main/java/com/facebook/presto/failureDetector/HeartbeatFailureDetector.java

```diff
@@ -143,6 +147,31 @@ public void shutdown()
                 .collect(toImmutableSet());
     }
 
+    @Override
+    public State getState(HostAddress hostAddress)
+    {
+        for (MonitoringTask task : tasks.values()) {
+            if (hostAddress.equals(HostAddress.fromUri(task.uri))) {
+                if (!task.isFailed()) {
+                    return State.ALIVE;
+                }
+                else {
+                    Exception lastException = task.getStats().getLastFailureException();
+                    if (lastException instanceof SocketTimeoutException || lastException instanceof UnknownHostException) {
+                        return State.GONE;
+                    }
+                    else if (lastException instanceof ConnectException) {
+                        return State.UNRESPONSIVE;
+                    }
+                    else {
+                        return State.UNKNOWN;
```

Comment:
Couple of things:
- Can we make this so it can look up hosts in a map? Not saying it's the best way, just curious.
- When does the unknown branch actually get taken? Should it ever happen? My guess is that it's super rare.